app/src/main/java/com/github/sdpteam15/polyevents/database/FirestoreDatabaseProvider.kt

```diff
@@ -203,37 +191,33 @@ object FirestoreDatabaseProvider : DatabaseInterface {
     }
 
     override fun getUserInformation(
-        user: Observable<UserInterface>,
-        uid: String,
-        userAccess: UserInterface
+        user: Observable<UserEntity>,
+        uid: String?,
+        userAccess: UserEntity?
     ): Observable<Boolean> = thenDoMultGet(
         firestore!!.collection(USER_COLLECTION)
-            .document(uid)
+            .document(uid!!)
             .get()
     ) {
         //TODO once the data class User is created, set the user with the correct value
         user.postValue(userAccess)
```

Comment:
Now you can create a new UserEntity based on the attributes returned by firebase right ? So that the user informations on the profile fragment will be load asynchronously

Reply:
Yes you're right! I didn't plan on doing it on this branch. I'll make a new branch for that when working on the other methods of the database that still need to be implemented :)